### PR TITLE
Update configuration wrt. using a projected shard

### PR DESCRIPTION
**Pull request by andreibancioiu**
- Add additional bootstrap files (for _projected shard_ = 0).
 - Define new `check:data` files for validating a rosetta instance started with `--observer-projected-shard = 0` (example).
 - Update `check:construction` configuration, as well.

Balances bootstrapping files were generated using [balancesExporter](https://github.com/ElrondNetwork/elrond-tools-go/tree/main/trieTools/balancesExporter):
 - for devnet using epochs 1165 and **1166**
 - for mainnet using epochs 702 and **703**

**Note:** the projected shard of an account is its containing shard, given a network with the maximum number of shards (256). In other words, the projected shard is given by the last byte of the public key.

### CODE REVIEW

File: checks/devnet/devnet-data-start-after-1399025-with-projected-shard.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "network": {
+    "blockchain": "Elrond",
+    "network": "D"
+  },
+  "http_timeout": 10,
+  "max_retries": 5,
+  "retry_elapsed_time": 0,
```
Comment (andreibancioiu) [inline checks/devnet/devnet-data-start-after-1399025-with-projected-shard.json:8]: These values will be updated when merging `feat/historical` (to easier keep track of changes).

File: checks/mainnet/mainnet-data-start-after-10120628-with-projected-shard.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
+  "network": {
+    "blockchain": "Elrond",
+    "network": "1"
+  },
+  "http_timeout": 10,
+  "max_retries": 5,
+  "retry_elapsed_time": 0,
```
Comment (andreibancioiu) [inline checks/mainnet/mainnet-data-start-after-10120628-with-projected-shard.json:8]: These values will be updated when merging `feat/historical` (to easier keep track of changes).

File: checks/devnet/bootstrap/D_shard_0(0)_epoch_1166_nonce_1399025_XeGLD.rosetta.json [hunk 3]
```diff
@@ -0,0 +1,1162 @@
+[
```
Comment (iulianpascalau) [inline checks/devnet/bootstrap/D_shard_0(0)_epoch_1166_nonce_1399025_XeGLD.rosetta.json:1]: can not check these values

File: checks/devnet/devnet-construction.json [hunk 4]
```diff
@@ -18,9 +18,9 @@
         "prefunded_accounts": [
             {
                 "account_identifier": {
-                    "address": "erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx"
+                    "address": "erd1ldjsdetjvegjdnda0qw2h62kq6rpvrklkc5pw9zxm0nwulfhtyqqtyc4vq"
                 },
-                "privkey": "b8ca6f8203fb4b545a8e83c5384da033c415db155b53fb5b8eba7ff5a039d639",
+                "privkey": "3e4e89e501eb542c12403fb15c52479e8721f2f4dedc3b3ef0f3b47b37de006c",
```
Comment (iulianpascalau) [inline checks/devnet/devnet-construction.json:23]: private keys in config? Do we really need this? Why?

Comment (andreibancioiu) [reply checks/devnet/devnet-construction.json:23]: We cannot use Rosetta's [`generate_key()`](https://www.rosetta-api.org/docs/rosetta_dsl_file.html#example-dsl-file) to generate keys in a given shard (whether actual shard or projected shard), therefore we have provide it both in `*.ros` and `*.json` config files. There are more like placeholders (even though I've prefunded these placeholder addresses, actually, on devnet) - these placeholders are generally replaced when testing.

**Review by miiu96 — APPROVED** [review]

File: checks/devnet/bootstrap/D_shard_0(0)_epoch_1166_nonce_1399025_XeGLD.rosetta.json (hunk 3 above)
Comment (andreibancioiu) [reply checks/devnet/bootstrap/D_shard_0(0)_epoch_1166_nonce_1399025_XeGLD.rosetta.json:1]: Balances bootstrapping files were generated using [balancesExporter](https://github.com/ElrondNetwork/elrond-tools-go/tree/main/trieTools/balancesExporter). The flow is described in the readme file. However, they can also be checked based on the existing bootstrap files:

```
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"reflect"

	"github.com/ElrondNetwork/elrond-go-core/core/pubkeyConverter"
	logger "github.com/ElrondNetwork/elrond-go-logger"
)

type rosettaBalance struct {
	AccountIdentifier rosettaAccountIdentifier `json:"account_identifier"`
	Currency          *rosettaCurrency         `json:"currency"`
	Value             string                   `json:"value"`
}

type rosettaAccountIdentifier struct {
	Address string `json:"address"`
}

type rosettaCurrency struct {
	Symbol   string `json:"symbol"`
	Decimals uint   `json:"decimals"`
}

func main() {
	projectedShard := byte(0)
	addressConverter, _ := pubkeyConverter.NewBech32PubkeyConverter(32, logger.GetOrCreate(""))
	dataWithActualShard, _ := ioutil.ReadFile("D_shard_0_epoch_1166_nonce_1399025_XeGLD.rosetta.json")
	dataWithProjectedShard, _ := ioutil.ReadFile("D_shard_0(0)_epoch_1166_nonce_1399025_XeGLD.rosetta.json")

	recordsWithActualShard := make([]rosettaBalance, 0)
	recordsWithProjectedShard := make([]rosettaBalance, 0)
	recordsWithProjectedShardRebuilt := make([]rosettaBalance, 0)
	_ = json.Unmarshal(dataWithActualShard, &recordsWithActualShard)
	_ = json.Unmarshal(dataWithProjectedShard, &recordsWithProjectedShard)

	for _, record := range recordsWithActualShard {
		publicKey, _ := addressConverter.Decode(record.AccountIdentifier.Address)
		if publicKey[len(publicKey)-1] == projectedShard {
			fmt.Println(record.AccountIdentifier.Address, record.Value)
			recordsWithProjectedShardRebuilt = append(recordsWithProjectedShardRebuilt, record)
		}
	}

	fmt.Println("Equal?", reflect.DeepEqual(recordsWithProjectedShardRebuilt, recordsWithProjectedShard))
}
```

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**